Gadm resolver all

ccjoel opened the pull request:
Draft of  changes for gadm resolver, compared to move-temporal-resolution (which is merged into this feature branch).

Pending:
- How/when to merge all transform changes with this, and merge order.

brandomr commented:
Captions are really implemented in Phantom via https://github.com/jataware/phantom/pull/108

mattprintz reviewed (approved):
Looks good

ccjoel committed:
Mock (in code) Gadm resolution step for Dojo dataset registration.

ccjoel committed:
Merge remote-tracking branch 'origin/gadm-resolver' into gadm-resolver-all

ccjoel committed:
CHeckpoint. Gadmresolver in the data transform panel on dataset registration.
Mocked job results so far.

ccjoel committed:
Some ntermediary cleanup.

ccjoel committed:
UI v2 implemented for gadm resolution within the transform step of data
transformation. Integration with backend (currently mock data) and changes from
feedback pending.

ccjoel committed:
CHanged gadm resolver table shadow to a thin border, and the table heading font
size was reduced.

ccjoel committed:
Gadm resolver UI making real rq job request to check for the data. Need to wire up the 'save' portion, clean up, and see what well do to implement the backend work as it is currently mocked.

ccjoel committed:
Merge remote-tracking branch 'origin/main' into gadm-resolver-all

ccjoel committed:
Merge remote-tracking branch 'origin/ui/move-temporal-resolution' into gadm-resolver-all

ccjoel committed:
CHanged schema of gadm detection, among other improvements.

ccjoel committed:
Almost done. Saved gadm resolution set by user. Updated stories. Need to change
owner of some state and will be in good shape.

ccjoel committed:
Gadm Overrides are reloaded from DataTransformation to GadmResolver when opening
and closing the panel.

ccjoel committed:
Removed duplicated required check for data transform review step- leftover from a merge.

Sorrento110 committed:
GADM resolver job updated.

Sorrento110 committed:
Merged with gadm resolver origin.

ccjoel committed:
Integrated GadmResolver with backend. Added some other exception handling for
when jobs finish too quicly with either success data on initial call, or a full
backend job_error (irrecoverable; where we display a stacktrace dialog for now).

ccjoel committed:
Merge remote-tracking branch 'origin/ui/move-temporal-resolution' into gadm-resolver-all

Sorrento110 committed:
Added in job to get all available gadm strings.

ccjoel committed:
Fetching list of all gadm countries in order to properly populate autocomplete.

ccjoel committed:
Cleaned up old code, old stories for it.

ccjoel committed:
Lint fixes.

ccjoel committed:
Handling lint ptoblems, removing stacktrace from errors from starting a elwood job.

Sorrento110 committed:
Look for the geo_type related to the admin_level.

ccjoel committed:
Adapted cypress dataset register tests to work with new data transform step
required operations.

ccjoel committed:
Merge branch 'gadm-resolver-all' of github.com:jataware/dojo into gadm-resolver-all

adamgilg commented on ui/client/datasets/DataTransformation/GadmResolver.js:
```diff
@@ -0,0 +1,275 @@
+  return (
+    <div
+      className={classes.root}
+      style={{ minHeight: 500 }}
```
This minHeight could be moved into the `root` class

adamgilg commented on ui/client/datasets/DataTransformation/DataTransformation.js:
```diff
@@ -257,6 +311,28 @@ const DataTransformation = withStyles(() => ({
+        setDataError('Nothing to review.');
+      }
+    } else {
+      setDataError('Something went wrong. Please contact Jataware for assitance.');
```
Typo: assitance -> assistance (both here and on line 331)

adamgilg commented on ui/client/datasets/DataTransformation/useElwoodData.js:
```diff
@@ -64,7 +65,27 @@ const useElwoodData = ({
 
       if (datasetId) {
-        startElwoodJob({ requestArgs }).then((jobId) => repeatFetch(jobId));
+        startElwoodJob({ requestArgs })
```
I'm going to pull this useElwoodData change into a new branch to start the work fixing the repeat call flow we talked about last week.

adamgilg replied:
Actually, I'm just going to bring in a skeleton version of this to avoid merge conflicts.

adamgilg commented:
This looks good! I haven't pulled it and tried it out locally yet but I'll give that a shot when it's ready for proper review.

adamgilg commented on ui/client/datasets/DataTransformation/useElwoodData.js:
```diff
@@ -9,6 +9,7 @@ const useElwoodData = ({
   generateArgs,
   jobString,
   cleanupRef,
+  onBackendFailure
```
`onBackendFailure` needs to be added as a dependency to the useEffect. I disabled the linter for the dependency array here because I didn't think I should be adding the ref to it, but on further exploration I believe it should be totally safe to add the ref without triggering any extra useEffect updates. So we can remove  `eslint-disable-next-line react-hooks/exhaustive-deps` from line 116 and add `cleanupRef` (and `onBackendFailure`).

ccjoel committed:
CHanged gadm overrides schema to allow data and to allow arbitrary child keys.

ccjoel committed:
Changing order of overrides and gadm key for gadm overrides.

ccjoel committed:
Using gadm overrides on the rqworker tasks side to pass in to elwood run_elwood job.

ccjoel committed:
Forgot to pass a paramater, fixing. FIxing regression that would occur only on
local dev.

ccjoel committed:
Configuring logger for two processors in order to inspect the info logs that are
already in place within gadm,elwood processors (rqworker).

ccjoel committed:
Ensure we update dataset annotations in react state so that run_elwood uses gadm
user overrides. this is on data transformation component.

ccjoel commented:
Closed, as created a new PR based against main.
